floss/language/go/extract.py

```diff
@@ -0,0 +1,385 @@
+    0048E6A0  78 01 09 46 74 72 75 6E  63 61 74 65 01 09 49 6E  x..Ftruncate..In
+    0048E6B0  74 65 72 66 61 63 65 01  09 4E 75 6D 4D 65 74 68  terface..NumMeth
+    0048E6C0  6F 64 01 09 50 72 65 63  69 73 69 6F 6E 01 09 52  od..Precision..R
+    """
+
+    blob_pattern = re.compile(b"(\x00|\x01)(?P<blob>.)", re.DOTALL)
+    for m in blob_pattern.finditer(section_data):
+        if m.group("blob") != b"\x00":
+            try:
+                addr = pe.get_offset_from_rva(m.start() + section_va + 2)
+                data = pe.get_data(m.end() + section_va, m.group(2)[0])
+
+                yield from decode_and_validate(data, addr, min_length)
+            except pefile.PEFormatError:
+                pass
+
+
+def extract_string_blob2(pe: pefile.PE, section_data, section_va, min_length) -> Iterable[StaticString]:
+    # Extract string blob in .rdata section that starts with "go:buildid" or "go.buildid"
+    """
+    67 6F 3A 62 75 69 6C 64  69 64 00 69 6E 74 65 72  go:buildid.inter
+    6E 61 6C 2F 63 70 75 2E  49 6E 69 74 69 61 6C 69  nal/cpu.Initiali
+    7A 65 00 69 6E 74 65 72  6E 61 6C 2F 63 70 75 2E  ze.internal/cpu.
+    70 72 6F 63 65 73 73 4F  70 74 69 6F 6E 73 00 69  processOptions.i
+    6E 74 65 72 6E 61 6C 2F  63 70 75 2E 69 6E 64 65  nternal/cpu.inde
+    78 42 79 74 65 00 69 6E  74 65 72 6E 61 6C 2F 63  xByte.internal/c
+    """
+
+    blob_pattern = re.compile(b"go(\.|:)buildid\x00(.)*\x00\x00", re.DOTALL)
+    for m in blob_pattern.finditer(section_data):
+        t = m.group(0)
+        length = 0
+        for s in t.split(b"\x00"):
+            addr = pe.get_offset_from_rva(m.start() + length + section_va)
+            length += len(s) + 1
+            yield from decode_and_validate(s, addr, min_length)
+
+
+def extract_string_blob_in_rdata_data(pe: pefile.PE, section_data, min_length, arch) -> Iterable[StaticString]:
+    # Extract strings from string table in .rdata section
+    # .data:00537B40                 dd offset unk_4A1E3C
+    # .data:00537B44                 dd    4
+    # .data:00537B48                 dd offset unk_4A21C2
+    # .data:00537B4C                 dd    6
+
+    if arch == "amd64":
+        alignment = 0x10
+        fmt = "<QQ"
+    else:
+        alignment = 0x8
+        fmt = "<II"
+
+    for addr in range(0, len(section_data) - alignment // 2, alignment // 2):
+        try:
+            curr = section_data[addr : addr + alignment]
+            s_off, s_size = struct.unpack(fmt, curr)
+
+            if not s_off and not (1 <= s_size < 128):
+                continue
+
+            s_rva = s_off - pe.OPTIONAL_HEADER.ImageBase
+
+            if not pe.get_section_by_rva(s_rva):
+                continue
+
+            binary_string = pe.get_string_at_rva(s_rva, s_size)
+
+            addr = pe.get_offset_from_rva(s_rva)
+
+            yield from decode_and_validate(binary_string, addr, min_length)
+
+        except Exception as e:
+            logger.error(f"Error: {e}")
+            raise
```

Comment:
use more specific exceptions and handle them (such as skipping invalid pointers and `continue`ing)